pkg/actors/actors.go

```diff
@@ -687,71 +687,62 @@ func (a *actorsRuntime) updateReminderTrack(actorKey, name string, repetition in
 	return err
 }
 
-func (a *actorsRuntime) getUpcomingReminderInvokeTime(reminder *Reminder) (time.Time, error) {
-	var nextInvokeTime time.Time
+func (a *actorsRuntime) startReminder(reminder *Reminder, stopChannel chan bool) error {
+	actorKey := constructCompositeKey(reminder.ActorType, reminder.ActorID)
+	reminderKey := constructCompositeKey(actorKey, reminder.Name)
 
-	registeredTime, err := time.Parse(time.RFC3339, reminder.RegisteredTime)
-	if err != nil {
-		return nextInvokeTime, errors.Wrap(err, "error parsing reminder registered time")
-	}
+	var (
+		nextInvokeTime, expirationTime time.Time
+		dueTime, period, ttl           time.Duration
+		repeats, repetitionsLeft       int
+	)
 
-	dueTime, err := time.ParseDuration(reminder.DueTime)
+	registeredTime, err := time.Parse(time.RFC3339, reminder.RegisteredTime)
 	if err != nil {
-		return nextInvokeTime, errors.Wrap(err, "error parsing reminder due time")
+		return errors.Wrap(err, "error parsing reminder registered time")
 	}
 
-	key := constructCompositeKey(reminder.ActorType, reminder.ActorID)
-	track, err := a.getReminderTrack(key, reminder.Name)
-	if err != nil {
-		return nextInvokeTime, errors.Wrap(err, "error getting reminder track")
+	if len(reminder.DueTime) != 0 {
+		if dueTime, err = time.ParseDuration(reminder.DueTime); err != nil {
+			return errors.Wrap(err, "error parsing reminder due time")
+		}
 	}
 
-	var lastFiredTime time.Time
-	if track != nil && track.LastFiredTime != "" {
-		lastFiredTime, err = time.Parse(time.RFC3339, track.LastFiredTime)
-		if err != nil {
-			return nextInvokeTime, errors.Wrap(err, "error parsing reminder last fired time")
+	repeats = -1
+	if len(reminder.Period) > 0 {
+		if period, repeats, err = parseDuration(reminder.Period); err != nil {
+			return errors.Wrap(err, "error parsing reminder period")
 		}
 	}
 
-	// the first execution reminder task
-	if lastFiredTime.IsZero() {
-		nextInvokeTime = registeredTime.Add(dueTime)
-	} else {
-		period, _, err := parseDuration(reminder.Period)
-		if err != nil {
-			return nextInvokeTime, errors.Wrap(err, "error parsing reminder period")
+	if len(reminder.TTL) > 0 {
+		if ttl, _, err = parseDuration(reminder.TTL); err != nil {
+			return errors.Wrap(err, "error parsing reminder TTL")
 		}
-		nextInvokeTime = lastFiredTime.Add(period)
+		expirationTime = registeredTime.Add(ttl).Add(dueTime)
```

Comment:
As discussed via Chat we have the opportunity to let someone specify the actual expiration Time via ISO 8601 on the period field.

Supporting the ISO8601 format with end time would give us this. This would allow for more flexibility as a user.